### ISSUE TITLE
Support cross-referencing equations

### ISSUE THREAD

**Issue by kaushalmodi** [kaushalmodi/ox-hugo#130]
See https://github.com/kaushalmodi/ox-hugo/issues/129.

**Comment by emacsun** [comment]
I use a kind of hybrid mode of the one-post-one-file  and one-post-one-subtree. Let me explain it for you.

I am a communication engineer, focusing the channel coding theory. So first, I want to write some summary about channel coding theory. Second, I also want to summarize my learning on math such as matrix, stochastic process and abstract linear. Third, I am very interested in computer science. So I think I can post some articles about computer science.

In summary, I have three section for my static site. Each has an Org file. i.e. math.org communication.org and computer.org. In each file, there are subtrees, each subtree for one post.

**Comment by kaushalmodi** [comment]
@emacsun 

I was reviewing your [comment in the original issue][1]:

> for equation, it will be like \ref{eq:XXXX} to refer the \label{eq:XXXX}

But looks like that's already supported. Do you have a test case that shows that that does not work?

[1]: https://github.com/kaushalmodi/ox-hugo/issues/129#issue-294089430

**Comment by kaushalmodi** [comment]
Test: https://ox-hugo.scripter.co/test/posts/links-to-equations/

I believe you need to add MathJax JS to the header for those `\ref` links to work. I add [these][1] to the `ox-hugo` test site HTML headers.

[1]: https://github.com/kaushalmodi/hugo-bare-min-theme/blob/master/layouts/partials/mathjax.html

**Comment by emacsun** [comment]
Hi @kaushalmodi , Here is a minimum case. the head of my telecommunication.org is ：

`
#+HUGO_BASE_DIR: ../
#+HUGO_SECTION: telecommunication
#+FILETAGS:telecommunication
#+SEQ_TODO: TODO NEXT DRAFT DONE
#+OPTIONS:   *:t <:nil timestamp:nil toc:t
#+HUGO_AUTO_SET_LASTMOD: t
** TODO Density Evolution             :LDPC:@telecommunication:
   :PROPERTIES:
   :EXPORT_FILE_NAME: density-evolution
   :EXPORT_DATE: <2018-02-05 Mon 14:10>
   :EXPORT_HUGO_CUSTOM_FRONT_MATTER+: :summary "Density evolution plays the foundamental role in designing and analyzing LDPC"
   :END:

It's easy to get:
\begin{eqnarray}
\label{eq:7}
\sum_{i}\lambda_{i}&=& 1 \newline
\sum_{i}\rho_{i} &=& 1
\end{eqnarray}

As mentioned in \ref{eq:7}

` 

In the webpage opened by chrome, the \ref{eq:7} is replace by ??? (yes, three ?)

Now, I am at company where the melpa cannot be visited. I will test the latest ox-hugo when I am home ASAP. :)

**Comment by emacsun** [comment]
Hi, @kaushalmodi 
Finally, I know why the corss-reference equation does not work for me. It is Hugo Academic theme. By default, the equation does not have a number at the right of the line.

So, after modify the mathjax config, it works!!!, Thank you @kaushalmodi very much.

`
      <script type="text/x-mathjax-config">
       MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] },
           displayAlign: "center",
                            displayIndent: "0em",
                            "HTML-CSS": { scale: 100,
                                          linebreaks: { automatic: "false" },
                                          webFont: "TeX"
                            },
                            SVG: {scale: 100,
                                  linebreaks: { automatic: "false" },
                                  font: "TeX"},
                            NativeMML: {scale: 100},
                            TeX: { equationNumbers: {autoNumber: "AMS"},
                                   MultLineWidth: "85%",
                                   TagSide: "right",
                                   TagIndent: ".8em"
                            }
       });
      </script>

`

**Comment by kaushalmodi** [comment]
Glad to hear! :) For reference, that MathJax config is what `ox-html` exports by default.

**Comment by kaushalmodi** [comment]
Actually, I'll leave this issue open till I document that the Org exported MathJax configuration is required.

**Comment by kaushalmodi** [comment]
Documented here: https://ox-hugo.scripter.co/doc/equations/
